tests/data_files/Makefile

```diff
@@ -1048,6 +1048,99 @@ cert_md5.crt: cert_md5.csr
+all_final += pkcs7-rsa-sha256-1.crt
+
+pkcs7-rsa-sha256-2.crt:
+	$(OPENSSL) req -x509 -subj="/C=NL/O=PKCS7/CN=PKCS7 Cert 2" -sha256 -nodes -days 365  -newkey rsa:2048 -keyout pkcs7-rsa-sha256-2.key -out pkcs7-rsa-sha256-2.crt
+	cat pkcs7-rsa-sha256-2.crt > pkcs7-rsa-sha256-2.pem
+	cat pkcs7-rsa-sha256-2.key >> pkcs7-rsa-sha256-2.pem
+all_final += pkcs7-rsa-sha256-2.crt
+
+#Generate data file to be signed
+pkcs7_data.txt:
+	echo "Hello" > $@
+	echo 2 >> pkcs7_data_1.txt
+all_final += pkcs7_data.txt
+
+#pkcs7 signature file with CERT
+pkcs7_data_cert.signed.sha256: pkcs7_data.txt pkcs7-rsa-sha256-1.crt
+	$(OPENSSL) smime -sign -binary -in pkcs7_data.txt -out $@ -md sha256 -signer pkcs7-rsa-sha256-1.pem -noattr -outform DER -out $@
+all_final += pkcs7_data_cert.signed.sha256
+
+#pkcs7 signature file with CERT and sha1
+pkcs7_data_cert.signed.sha1: pkcs7_data.txt
+	$(OPENSSL) smime -sign -binary -in pkcs7_data.txt -out $@ -md sha1 -signer pkcs7-rsa-sha256-1.pem -noattr -outform DER -out $@
+all_final += pkcs7_data_cert.signed.sha1
+
+#pkcs7 signature file with CERT and sha512
+pkcs7_data_cert.signed.sha512: pkcs7_data.txt
+	$(OPENSSL) smime -sign -binary -in pkcs7_data.txt -out $@ -md sha512 -signer pkcs7-rsa-sha256-1.pem -noattr -outform DER -out $@
+all_final += pkcs7_data_cert.signed.sha512
+
+#pkcs7 signature file without CERT
+pkcs7_data_without_cert.signed: pkcs7_data.txt
+	$(OPENSSL) smime -sign -binary -in pkcs7_data.txt -out $@ -md sha256 -signer pkcs7-rsa-sha256-1.pem -nocerts -noattr -outform DER -out $@
+all_final += pkcs7_data_without_cert.signed
+
+#pkcs7 signature file with multiple signers
+pkcs7_data_multiple.signed: pkcs7_data.txt
+	$(OPENSSL) smime -sign -binary -in pkcs7_data.txt -out $@ -md sha256 -signer pkcs7-rsa-sha256-1.pem -signer pkcs7-rsa-sha256-2.pem -nocerts -noattr -outform DER -out $@
+all_final += pkcs7_data_multiple.signed
+
+#pkcs7 signature file with multiple certificates
+pkcs7_data_multiple_certs.signed: pkcs7_data.txt
+	$(OPENSSL) smime -sign -binary -in pkcs7_data.txt -out $@ -md sha256 -signer pkcs7-rsa-sha256-1.pem -signer pkcs7-rsa-sha256-2.pem -noattr -outform DER -out $@
+all_final += pkcs7_data_multiple_certs.signed
+
+#pkcs7 signature file with corrupted CERT
+pkcs7_data.signed.badcert: pkcs7_data.txt
+	cp pkcs7_data_cert.signed.sha256 $@
+	echo -en '\xa1' | dd of=$@ bs=1 seek=547 conv=notrunc
+all_final += pkcs7_data.signed.badcert
+
+#pkcs7 signature file with corrupted signer info
+pkcs7_data.signed.badsigner: pkcs7_data.txt
+	cp pkcs7_data_cert.signed.sha256 $@
+	echo -en '\xa1' | dd of=$@ bs=1 seek=918 conv=notrunc
+all_final += pkcs7_data.signed.badsigner
+
+#pkcs7 file with CRL
+pkcs7_data_crl.signed:
+		cat pkcs7-rsa-sha256-1.pem > pkcs7-rsa-crt-crl.pem
+		cat crl_sha256.pem >> pkcs7-rsa-crt-crl.pem
+		$(OPENSSL) smime -sign -binary -in pkcs7_data.txt -out $@ -md sha512 -signer pkcs7-rsa-crt-crl.pem -inkey pkcs7-rsa-sha256-1.key -noattr -outform DER -out $@
```

Comment:
If you depend on `crl_sha256.pem` you should probably add that to the top line (`pkcs7_data_crl.signed: crl_sha256.pem`). There are a few dependencies like this in your changes.